python/metatensor_torch/metatensor/torch/serialization.py

```diff
@@ -0,0 +1,159 @@
+import os
+import pathlib
+from typing import BinaryIO, Union
+
+import numpy as np
+import torch
+
+
+if os.environ.get("METATENSOR_IMPORT_FOR_SPHINX", "0") != "0":
+    from .documentation import (
+        Labels,
+        LabelsEntry,
+        TensorBlock,
+        TensorMap,
+    )
+else:
+    Labels = torch.classes.metatensor.Labels
+    LabelsEntry = torch.classes.metatensor.LabelsEntry
+    TensorBlock = torch.classes.metatensor.TensorBlock
+    TensorMap = torch.classes.metatensor.TensorMap
+
+
+# the functions in this file are written in a relatively strange way to enable them
+# taking `Path` and `BinaryIO` arguments when running in Python mode, while they only
+# support str in TorchScript mode
+
+
+def load(file: str) -> TensorMap:
+    """
+    Load a previously saved :py:class:`TensorMap` from the given ``file``.
+
+    :py:class:`TensorMap` are serialized using the ``.mts`` format, i.e. a ZIP file
+    without compression (storage method is ``STORED``), where each file is stored as a
+    ``.npy`` array. See the C API documentation for more information on the format.
+
+    :param file: path of the file to load, or already opened file.
+
+        .. warning::
+
+            When using this function in TorchScript mode, only ``str`` arguments are
+            supported.
+    """
+    if torch.jit.is_scripting():
+        assert isinstance(file, str)
+        return torch.ops.metatensor.load(file=file)
+    else:
+        if isinstance(file, str):
+            return torch.ops.metatensor.load(file=file)
+        elif isinstance(file, pathlib.Path):
+            return torch.ops.metatensor.load(file=str(file))
+        else:
+            # assume a file-like object
+            buffer = file.read()
+            assert isinstance(buffer, bytes)
+
+            buffer = torch.tensor(np.frombuffer(buffer, dtype=np.uint8))
```

Comment:
```suggestion
            buffer = torch.frombuffer(buffer, dtype=torch.uint8)
```

Also why hard-code the `dtype`?